src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "Device/Driver/AR62xx.hpp"
+#include "Device/Driver.hpp"
+#include "Device/Port/Port.hpp"
+#include "NMEA/Info.hpp"
+#include "RadioFrequency.hpp"
+#include "Thread/Cond.hxx"
+#include "Thread/Mutex.hxx"
+#include "Util/CharUtil.hxx"
+#include "Util/StaticFifoBuffer.hxx"
+#include "Util/Compiler.h"
+#include "LogFile.hpp"
+
+#include <cstdint>
+
+#include <stdio.h>
+
+/** max command length to send to AR62xx*/
+const int MAX_CMD_LEN = 128;
+
+/**
+ * active / passive station
+ * the ar62xx can hold 2 stations in parallel
+ * 1 = active station of AR62xx
+ * 0 = passive station of AR62xx
+ */
+const int ACTIVE_STATION = 1;
+const int PASSIVE_STATION = 0;
+
+/** the lowest frequency which can be set in AR62xx */
+const unsigned MIN_FREQUENCY = 118000;
+
+/** the highest frequency which can be set in AR62xx */
+const unsigned MAX_FREQUENCY = 137000;
+
+/** the range of all available frequencies */
+const unsigned FREQUENCY_RANGE = MAX_FREQUENCY - MIN_FREQUENCY;
+
+/** 
+ * max frequency id
+ * the highest frequency ID which can be set in AR62xx.
+ * this number is used to calculate the particular frequency-id of each frequency.
+ * 
+ * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 kHz
+ * e.g. frequency-id 0    => 118.000 kHz
+ *      frequency-id 1    => 118.005 kHz
+ *      frequency-id 3040 => 137.000 kHz
+ */
+const int MAX_FREQUENCY_ID = 3040;
+
+/** bitmasks to get the frequency out of a frequency-id */
+const int FREQUENCY_BITMASK = 0xFFF0;
+
+/** bitmasks to get the channel out of a frequency-id */
+const int CHANNEL_BITMASK = 0xF;
+
+class AR62xxDevice final : public AbstractDevice
+{
+
+public:
+  explicit AR62xxDevice(Port &_port);
+
+private:
+  /** Port the radio is connected to. */
+  Port &port;
+
+  /** active station frequency*/
+  RadioFrequency active_frequency;
+
+  /** passive (or standby) station frequency*/
+  RadioFrequency passive_frequency;
+
+  /**
+   * @brief Send - Sends a message to the AR62xx
+   * 
+   * @param msg message
+   * @param msg_size size of the message
+   * @param env operation environment
+   * @return true if message has been sent
+   * @return false if message could not be sent
+   */
+  bool Send(const uint8_t *msg, unsigned msg_size, OperationEnvironment &env);
+
+  /**
+   * @brief converts a frequency to an AR62xx-ID
+   * 
+   * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 kHz
+   * e.g. frequency-id 0    => 118.000 kHz
+   *      frequency-id 1    => 118.005 kHz
+   *      frequency-id 3040 => 137.000 kHz
+   * 
+   * @param frequency frequency in kHz
+   * @return uint16_t frequency-id
+   */
+  uint16_t FrequencyToAR62xxId(RadioFrequency frequency);
```

Comment:
This method needs a non-`const` instance, but why?

Reply:
set to const